tests_tf/test_attacks.py

```diff
@@ -250,6 +268,69 @@ def test_generate_np_clip_works_as_expected(self):
     self.assertClose(np.min(x_adv), -0.2)
     self.assertClose(np.max(x_adv), 0.1)
 
+class TestFastGradientMethod(CommonAttackProperties):
+
+  def setUp(self):
+    super(TestFastGradientMethod, self).setUp()
+
+    self.attack = FastGradientMethod(self.model, sess=self.sess)
+
+  def test_optimize_linear_l1(self):
```

Comment:
I think it would be better to put `test_optimize_linear_l1_ties` and `test_optimize_linear_l1` into separate test class.
These two tests are testing `optimize_linear` helper function, which is not specific to FGSM attack.
Also these two test do not utilize any methods declared in `CommonAttackProperties`

You can have a class which is called, for examples, `TestAttackHelpers` which would contain tests for `optimize_linear` and potentially other helper methods:

```
class TestAttackHelpers(CleverHansTest)):

  def test_optimize_linear_l1(self):
    ...

  def test_optimize_linear_l1_ties(self):
    ...
```


Also, I think it make sense to add tests for other values of `ord` in the `optimize_linear`.
While they not necessarily have special corner cases, it's useful to test them in case somebody will be doing refactoring and introduce some error in the code.
Direct tests for `optimize_linear` (instead of relying on tests for PGD, FGSM, etc...) will make finding such errors easier.

Reply:
Done